In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

import pandas_profiling as pp

from sklearn.feature_selection import VarianceThreshold, SelectKBest, SelectFromModel
from sklearn.linear_model import LinearRegression
from sklearn.svm import LinearSVC
from sklearn.model_selection import train_test_split

from mlxtend.feature_selection import SequentialFeatureSelector as SFS


pd.set_option("display.max_columns",999 )

In [14]:
# This is paribas clain management data set
# Importing data sets into pandas
# importing 100000 for perfemance improvment
data = pd.read_csv('train.csv', nrows=100000)

In [21]:
data.info()
# data.head()
# data.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Columns: 133 entries, ID to v131
dtypes: float64(108), int64(6), object(19)
memory usage: 101.5+ MB


### Let us do pre-processing of the dataset

#### Find Null value columns


In [16]:
# find null valus

null_values =   [var for var in data.columns if data[var].isnull().sum()>0]
print('Total number of null value columns', len(null_values))

Total number of null value columns 119


We saw out of 133 features 119 columns are having null values. Let us see the percentage of null values.

In [28]:
data[null_values].isnull().sum()

v1      43552
v2      43517
v3       3027
v4      43517
v5      42542
v6      43552
v7      43552
v8      42537
v9      43569
v10        74
v11     43556
v12        76
v13     43552
v14         4
v15     43556
v16     43609
v17     43517
v18     43552
v19     43562
v20     43560
v21       528
v22       439
v23     44298
v25     42537
v26     43552
v27     43552
v28     43552
v29     43552
v30     52645
v31      3027
        ...  
v100    43556
v101    43517
v102    44852
v103    43552
v104    43552
v105    42565
v106    43517
v107        3
v108    42542
v109    42542
v111    43552
v112      334
v113    48319
v114       26
v115    43609
v116    43556
v117    42542
v118    43562
v119    44303
v120    43556
v121    43560
v122    43569
v123    44301
v124    42537
v125       68
v126    43552
v127    43552
v128    42542
v130    43562
v131    43609
Length: 119, dtype: int64

In [35]:
# Let us 
data.dropna(axis = 0, how = 'any')

In [36]:
dataset.shape

(15556, 133)

In [53]:
# Check if our data contains null values
cols = [col for col in data.columns if data[col].isnull().sum() > 0]
print(np.size(cols))

119


In [23]:
data['target'].unique()

array([1, 0], dtype=int64)

In [24]:
data.get_dtype_counts()

float64    108
int64        6
object      19
dtype: int64

In [35]:
# Splitting data into train and test sets
#
# Set the random state while splitting data into train
# and test set, as that ensures the reproducibility

X_train, X_test, y_train, y_test = train_test_split(data.drop('target', axis=1),data['target'],test_size = 0.3, random_state = 0)

In [44]:
numerical_cols = [var for var in X_train.columns if 'float64' == data[var].dtype or 'int64' == data[var].dtype]

In [45]:
X_train[numerical_cols].dtypes.value_counts()

float64    108
int64        5
dtype: int64

In [48]:
# Let us find the constant columns
# from the neumerical columns
# by calculating the standard deviation

constant_columns = [var for var in X_train[numerical_cols].columns if data[var].std()==0]
constant_columns

[]

Here we don't find a constand column. Let us look for quasi constant features

In [49]:
# Finding quasi constant features based
# threasold value

# assigning thresold value
threshold=0.98

def get_quasi_constant(data, thresold):
    dataset = data.copy().dropna()
    sel = VarianceThreshold(threshold=thresold)
    sel.fit(dataset[numerical_cols])
    constant_columns = [var for var in dataset[numerical_cols].columns  if var not is in dataset[numerical_cols].columns[sel.get_support()]]
    
quasi_constant_features = get_quasi_constant(data,threshold)

SyntaxError: invalid syntax (<ipython-input-49-b1bce5a988cc>, line 11)

VarianceThreshold(threshold=0.98)

SyntaxError: invalid syntax (<ipython-input-66-ad38479b7c64>, line 1)

In [49]:
constant_columns

['ID',
 'v1',
 'v2',
 'v4',
 'v5',
 'v8',
 'v9',
 'v13',
 'v16',
 'v17',
 'v20',
 'v23',
 'v25',
 'v28',
 'v29',
 'v34',
 'v35',
 'v36',
 'v39',
 'v40',
 'v44',
 'v45',
 'v46',
 'v48',
 'v50',
 'v51',
 'v54',
 'v58',
 'v59',
 'v61',
 'v63',
 'v64',
 'v65',
 'v67',
 'v68',
 'v69',
 'v70',
 'v76',
 'v77',
 'v78',
 'v80',
 'v81',
 'v82',
 'v83',
 'v85',
 'v87',
 'v88',
 'v89',
 'v93',
 'v96',
 'v97',
 'v98',
 'v100',
 'v101',
 'v102',
 'v103',
 'v105',
 'v106',
 'v109',
 'v111',
 'v114',
 'v115',
 'v117',
 'v118',
 'v119',
 'v121',
 'v122',
 'v123',
 'v124',
 'v127',
 'v130',
 'v131']

In [1]:
dataset['v118'].std()

NameError: name 'dataset' is not defined

In [59]:
%lsmagic

Available line magics:
%alias  %alias_magic  %autoawait  %autocall  %automagic  %autosave  %bookmark  %cd  %clear  %cls  %colors  %config  %connect_info  %copy  %ddir  %debug  %dhist  %dirs  %doctest_mode  %echo  %ed  %edit  %env  %gui  %hist  %history  %killbgscripts  %ldir  %less  %load  %load_ext  %loadpy  %logoff  %logon  %logstart  %logstate  %logstop  %ls  %lsmagic  %macro  %magic  %matplotlib  %mkdir  %more  %notebook  %page  %pastebin  %pdb  %pdef  %pdoc  %pfile  %pinfo  %pinfo2  %popd  %pprint  %precision  %prun  %psearch  %psource  %pushd  %pwd  %pycat  %pylab  %qtconsole  %quickref  %recall  %rehashx  %reload_ext  %ren  %rep  %rerun  %reset  %reset_selective  %rmdir  %run  %save  %sc  %set_env  %store  %sx  %system  %tb  %time  %timeit  %unalias  %unload_ext  %who  %who_ls  %whos  %xdel  %xmode

Available cell magics:
%%!  %%HTML  %%SVG  %%bash  %%capture  %%cmd  %%debug  %%file  %%html  %%javascript  %%js  %%latex  %%markdown  %%perl  %%prun  %%pypy  %%python  %%python2  %%